xcube_stac/store.py

```diff
@@ -97,10 +118,73 @@ def get_item_collection(
         using the supplied *open_params*.
 
         Returns:
-            A tuple of the item collection containing all items
-            identified by *open_params* and data IDs corresponding to items
+            items: item collection containing all items identified by *open_params*
+            item_data_ids: data IDs corresponding to items
+        """
+        if self._searchable:
+            # not used
+            open_params.pop("variable_names", None)
+            # rewrite to "datetime"
+            time_range = open_params.pop("time_range", None)
+            if time_range:
+                open_params["datetime"] = "/".join(time_range)
+            items = self._catalog.search(**open_params).item_collection()
+        else:
+            items = self._get_items_nonsearchable_catalog(
+                self._catalog,
+                **open_params
+            )
+            items = pystac.ItemCollection(items)
+        item_data_ids = self.list_item_data_ids(items)
+        return items, item_data_ids
+
+    def get_item_data_id(self, item: pystac.Item) -> str:
+        """Generates the data ID of an item, which follows the structure:
+
+            `collection_id_0/../collection_id_n/item_id`
+
+        Args:
+            item: item/feature
+
+        Returns:
+            data ID of an item
+        """
+        id_parts = []
+        parent_item = item
+        while parent_item.STAC_OBJECT_TYPE != "Catalog":
+            id_parts.append(parent_item.id)
+            parent_item = parent_item.get_parent()
+        id_parts.reverse()
+        return self._data_id_delimiter.join(id_parts)
+
+    def get_item_data_ids(self, items: Iterable[pystac.Item]) -> Iterator[str]:
```

Comment:
Should not be public.

Reply:
This method will be removed with the next PR. New data IDs don`t need this method.